frontend/src/util/Hook.tsx

```diff
@@ -37,37 +38,39 @@ const useGetScore = (player?: Player) => {
           counted.add(player.submissions[i].problemIndex);
         }
       }
+
+      setScore(counted.size);
     }
-  }, [player]);
+    
+  }, [player, setScore]);
 
   if (player == null || player.submissions.length === 0) {
     return null;
   }
-
-  return counted.size;
+  
+  return score;
 };
 
 export default useGetScore;
 
 export const useGetSubmissionTime = (player?: Player) => {
-  const counted = new Set<Number>();
-  let time;
+  const counted = new Set<number>();
+  const [time, setTime] = useState<string>();
 
   useEffect(() => {
     if (player) {
       for (let i = 0; i < player.submissions.length; i += 1) {
-        if (player.submissions[i].submissionType === SubmissionType.Submit &&
-          player.submissions[i].numCorrect === player.submissions[i].numTestCases &&
+        if (player.submissions[i].numCorrect === player.submissions[i].numTestCases &&
           !counted.has(player.submissions[i].problemIndex)) {
           counted.add(player.submissions[i].problemIndex);
-          time = player.submissions[i].startTime;
+          setTime(player.submissions[i].startTime);
```

Comment:
I'm sort of confused - is there a difference between correct and best submissions? Idk how it plays into effect when there are multiple problems though, since from this code I get the impression that it finds the time of the latest correct submission of a problem that hasn't been solved yet (idk if this is actually true though esp. with a non-state set variable). Maybe @jasonz6688 can clarify after this is merged?

Reply:
The code should find the time of the last new correct submission. I can explain it in call, but I'm pretty confident in what it does.

Best submission is just legacy code I left in there so that we can modify it for the end game code. But best submission theoretically finds the submission that solved the most test cases. Correct just means that a submission solved all the test cases for the problem.